enterprise/server/remote_execution/commandutil/commandutil_windows.go

```diff
@@ -0,0 +1,89 @@
+//go:build windows
+
+package commandutil
+
+import (
+	"fmt"
+	"os/exec"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+type process struct {
+	cmd        *exec.Cmd
+	terminated chan struct{}
+
+	jobHandle windows.Handle
+}
+
+func (p *process) preStart() error {
+	j, err := createWindowsJobObject()
+	if err != nil {
+		return err
+	}
+
+	p.jobHandle = j
+	return nil
+}
+
+func (p *process) postStart() error {
+	processHandle, err := windows.OpenProcess(windows.PROCESS_QUERY_LIMITED_INFORMATION|windows.PROCESS_TERMINATE, false, uint32(p.cmd.Process.Pid))
+	if err != nil {
+		return nil
+	}
+	defer windows.CloseHandle(processHandle)
+
+	return windows.AssignProcessToJobObject(p.jobHandle, processHandle)
+}
+
+func createWindowsJobObject() (windows.Handle, error) {
+	job, err := windows.CreateJobObject(nil, nil)
+	if err != nil {
+		return windows.InvalidHandle, fmt.Errorf("failed to create job object: %w", err)
+	}
+	limitInfo := windows.JOBOBJECT_BASIC_LIMIT_INFORMATION{
+		LimitFlags: windows.JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE,
+	}
+	if _, err := windows.SetInformationJobObject(
+		job,
+		windows.JobObjectBasicLimitInformation,
+		uintptr(unsafe.Pointer(&limitInfo)),
+		uint32(unsafe.Sizeof(limitInfo))); err != nil {
+		return windows.InvalidHandle, fmt.Errorf("failed to set job object info: %w", err)
+	}
+
+	return job, nil
+}
+
+// killProcessTree kills the given pid as well as any descendant processes.
+//
+// It tries to kill as many processes in the tree as possible. If it encounters
+// an error along the way, it proceeds to kill subsequent pids in the tree. It
+// returns the last error encountered, if any.
+//
```

Comment:
nit: remove this part since this is more describing the unix-specific behavior

Reply:
Done